src/chapterintrofx.c

```diff
@@ -0,0 +1,1344 @@
+
+    sub_8001710();
+
+    sub_800172C(gPaletteBuffer, 0, 3, -2);
+    sub_800172C(gPaletteBuffer + 0x40, 4, 2, -2);
+    sub_800172C(gPaletteBuffer + 0xE0, 0xE, 2, -2);
+    sub_800172C(gPaletteBuffer + 0x120, 0x12, 1, -2);
+
+    proc->unk_4C = 0xF;
+
+    Sound_FadeOutSE(1);
+
+    return;
+}
+
+void ChapterIntro_LoopFadeOut(struct ChapterIntroFXProc* proc) {
+    sub_80D74B0();
+    EnablePaletteSync();
+
+    proc->unk_4C--;
+
+    if (proc->unk_4C < 0) {
+        gLCDControlBuffer.dispcnt.bg0_on = 0;
+        gLCDControlBuffer.dispcnt.bg1_on = 0;
+        gLCDControlBuffer.dispcnt.bg2_on = 0;
+        gLCDControlBuffer.dispcnt.bg3_on = 0;
+        gLCDControlBuffer.dispcnt.obj_on = 0;
+
+        SetBackgroundTileDataOffset(2, 0);
+        Proc_Break(proc);
+    }
+
+    return;
+}
+
+void ChapterIntro_BeginFastFadeToMap(struct ChapterIntroFXProc* proc) {
+    gLCDControlBuffer.dispcnt.mode = 0;
+
+    CpuFastFill(0, (void *)BG_VRAM, 32);
+
+    BG_Fill(gBG0TilemapBuffer, 0);
+    BG_Fill(gBG1TilemapBuffer, 0);
+    BG_Fill(gBG2TilemapBuffer, 0);
+
+    BG_EnableSyncByMask(7);
+
+    Proc_EndEach(sProcScr_ChapterIntro_Bg3Scroll);
+    Proc_EndEach(sProcScr_ChapterIntro_0859B108);
+    Proc_EndEach(sProcScr_ChapterIntro_0859B160);
+
+    sub_8001710();
+
+    sub_800172C(gUnknown_02022968, 6, 10, 2);
+    sub_800172C(gUnknown_02022968 + 0x140, 0x1A, 6, 2);
+    sub_800172C(gUnknown_02022968 + 0xA0, 0x10, 2, 2);
+    sub_800172C(gUnknown_02022968 + 0x110, 0x17, 1, 2);
+
+    sub_80D74B0();
+
+    EnablePaletteSync();
+
+    proc->unk_4C = 0xE;
+
+    if ((GetROMChapterStruct(gRAMChapterData.chapterIndex)->unk28) != 0xFFFF) {
+        Sound_PlaySong80024D4(GetROMChapterStruct(gRAMChapterData.chapterIndex)->unk28, 0);
+    }
+
+    return;
+}
+
+void ChapterIntro_LoopFastFadeToMap(struct ChapterIntroFXProc* proc) {
+    sub_80D74B0();
+
+    if ((GetROMChapterStruct(gRAMChapterData.chapterIndex)->initialWeather) == 5) {
+        WfxFlamesInitGradientPublic();
+    }
+
+    if ((GetChapterThing() == 2) || (GetROMChapterStruct(gRAMChapterData.chapterIndex)->boolFadeToBlack)) {
+        proc->unk_4C = 0;
+
+        gLCDControlBuffer.dispcnt.bg0_on = 1;
+        gLCDControlBuffer.dispcnt.bg1_on = 1;
+        gLCDControlBuffer.dispcnt.bg2_on = 1;
+        gLCDControlBuffer.dispcnt.bg3_on = 0;
+        gLCDControlBuffer.dispcnt.obj_on = 0;
+    } else {
+        EnablePaletteSync();
+    }
+
+    proc->unk_4C--;
+
+    if (proc->unk_4C < 0) {
+        ResetMapPaletteAnimations();
+        Proc_Break(proc);
+    }
+
+    return;
+}
+
+void ChapterIntro_SetSkipTarget(s16 arg, struct ChapterIntroFXProc* proc) {
+    proc->unk_50 = arg;
+
+    return;
+}
+
+
+void ChapterIntro_SetTimerMaybe(s16 arg, struct ChapterIntroFXProc* proc) {
+    proc->unk_4C = arg;
+
+    return;
+}
+
+#if NONMATCHING
+
+void ChapterIntro_TickTimerMaybe(struct ChapterIntroFXProc* proc) {
+    if (proc->unk_52 != 0) {
+        Proc_Break(proc);
+        return;
+    }
```

Comment:
Is the return here required? Maybe move the decrement into the if and make it an else if instead of using a return?